### PR TITLE
Marvell 88F7040 drivers porting for Linux Kernel 5.10 - DNI boards

### PR DESCRIPTION
**Pull request by eladnachman**
This commit contains the complete 4 part series of Linux Kernel 5.10 patches required for Marvell 88F7040 SOC drivers porting for DNI boards, and modification needed in the arm64 .config file to enable the Marvell 10G PHY driver.

Signed-off-by: Elad Nachman <enachman@marvell.com>

### CODE REVIEW

**Review by taraschornyiplv — APPROVED** [review]

File: packages/base/any/kernels/5.10-lts/patches/0024-i2c-Added-mv64xxx-support-for-AXI-bus-clock-handling.patch [hunk 1]
```diff
@@ -0,0 +1,109 @@
+From 5604c64109bcd315c5d2e5461b7304f361d93642 Mon Sep 17 00:00:00 2001
+From: Elad Nachman <enachman@marvell.com>
+Date: Wed, 28 Jul 2021 13:00:30 +0300
+Subject: [PATCH 1/4] i2c: Added mv64xxx support for AXI bus clock handling
```
Comment (paulmenzel) [inline packages/base/any/kernels/5.10-lts/patches/0024-i2c-Added-mv64xxx-support-for-AXI-bus-clock-handling.patch:4]: Please use imperative mood presence. [https://chris.beams.io/posts/git-commit/](https://chris.beams.io/posts/git-commit/)

File: packages/base/any/kernels/5.10-lts/patches/0024-i2c-Added-mv64xxx-support-for-AXI-bus-clock-handling.patch [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+ 	if (!IS_ERR(drv_data->reg_clk))
+ 		clk_prepare_enable(drv_data->reg_clk);
+ 
++	drv_data->axi_clk = devm_clk_get(&pd->dev, "axi");
++	if (IS_ERR(drv_data->axi_clk) &&
++	    PTR_ERR(drv_data->axi_clk) == -EPROBE_DEFER) {
++		return -EPROBE_DEFER;
++	}
++	if (!IS_ERR(drv_data->axi_clk))
++		clk_prepare_enable(drv_data->axi_clk);
```
Comment (paulmenzel) [inline packages/base/any/kernels/5.10-lts/patches/0024-i2c-Added-mv64xxx-support-for-AXI-bus-clock-handling.patch:86]: Please follow the Linux kernel coding style, and run the patch through `checkpatch.pl`.

File: packages/base/any/kernels/5.10-lts/patches/0024-i2c-Added-mv64xxx-support-for-AXI-bus-clock-handling.patch [hunk 3]
```diff
@@ -0,0 +1,109 @@
+From 5604c64109bcd315c5d2e5461b7304f361d93642 Mon Sep 17 00:00:00 2001
```
Comment (paulmenzel) [inline packages/base/any/kernels/5.10-lts/patches/0024-i2c-Added-mv64xxx-support-for-AXI-bus-clock-handling.patch:1]: This commit is not in the Linus’ Linux master branch. It should be upstreamed first, and then backported.

File: packages/base/any/kernels/5.10-lts/patches/0025-net-Added-MII-support-to-mvpp2-driver.patch [hunk 4]
```diff
@@ -0,0 +1,132 @@
+From 67cbf15c026cec3887e19d02bd82164ecfdd697a Mon Sep 17 00:00:00 2001
+From: Elad Nachman <enachman@marvell.com>
+Date: Wed, 28 Jul 2021 13:34:39 +0300
+Subject: [PATCH 2/4] net: Added MII support to mvpp2 driver
+
+This patch adds MII support to the Marvell PPv2
+(Packet Processor v2) 1/10 Gbps adapter driver.
```
Comment (paulmenzel) [inline packages/base/any/kernels/5.10-lts/patches/0025-net-Added-MII-support-to-mvpp2-driver.patch:7]: 1.  Please re-flow for 75 characters per line.
2.  Please mention the datasheet name and revision.

**Review by paulmenzel — CHANGES_REQUESTED** [review]
Can you make it four or five commits too (one for each patch and one for the configuration change)?

To avoid the maintenance burden known from Open Network Linux (ONL), the Linux kernel patches need to be sent for review upstream (so based on Linux master first) possibly with Cc: stable@vger.kernel.org.

**Pull request closed without merging**